src/plugins/intel_cpu/src/nodes/interaction.cpp

```diff
@@ -0,0 +1,220 @@
+    for (uint32_t n = 0; n < inputSizes; n++) {
+        auto inPtr = reinterpret_cast<const uint8_t*>(getParentEdgeAt(n)->getMemoryPtr()->GetPtr());
+        inputPtrs[n] = inPtr;
+    }
+    for (int64_t start = 0; start < batchSize; start++) {
+        cat(inputPtr->buffer().as<uint8_t*>(), inputPtrs, featureSizes, start, dataPrecision.size());
+        std::unordered_map<int, memory> mem_ags {
+            {DNNL_ARG_SRC, inputMemPtr->GetPrimitive()},
+            {DNNL_ARG_WEIGHTS, inputMemPtr->GetPrimitive()},
+            {DNNL_ARG_DST, outputMemPtr->GetPrimitive()}
+        };
+        (*prim).execute(strm, mem_ags);
+        flat_triangle(outputPtr->buffer().as<uint8_t*>(),
+            flatPtr->buffer().as<uint8_t*>(), inputSizes, dataPrecision.size());
+        //in1 dense feature
+        //in2 flatted interaction features
+        cat(
+          inputPtrs[0] + start * featureSize * dataPrecision.size(),
+          flatPtr->buffer().as<uint8_t*>(),
+          outFeaturesPtr + start * outputFeaturesLen * dataPrecision.size(),
+          featureSize,
+          interactFeatureSize,
+          dataPrecision.size());
+    }
+}
+
+
+
+void Interaction::execute(dnnl::stream strm) {
+    execRef(strm);
+}
+
+bool Interaction::created() const {
+    return getType() == Type::Interaction;
+}
+
+void Interaction::prepareParams() {
+    using tag = dnnl::memory::format_tag;
+    using dt = dnnl::memory::data_type;
+    using namespace dnnl;
+    const auto& denseFeatureDims = getParentEdgeAt(0)->getMemory().getStaticDims();
+    batchSize = denseFeatureDims[0];
+    featureSize = denseFeatureDims[1];
+    inputSizes = inputShapes.size();
+    interactFeatureSize = inputSizes * (inputSizes - 1) / 2;
+    outputFeaturesLen = interactFeatureSize + featureSize;
+    std::vector<int64_t> lhsShape({static_cast<int64_t>(inputSizes), static_cast<int64_t>(featureSize)});
+    std::vector<int64_t> lhsStride({static_cast<int64_t>(featureSize), 1});
+    std::vector<int64_t> rhsShape({static_cast<int64_t>(featureSize), static_cast<int64_t>(inputSizes)});
+    std::vector<int64_t> rhsStride({1, static_cast<int64_t>(featureSize)});
+    std::vector<int64_t> resShape({static_cast<int64_t>(inputSizes), static_cast<int64_t>(inputSizes)});
+    std::vector<int64_t> resStride({static_cast<int64_t>(inputSizes), 1});
+    auto dataType = DnnlExtensionUtils::IEPrecisionToDataType(dataPrecision);
+    auto src_md = memory::desc(lhsShape, dataType, lhsStride);
+    auto weights_md = memory::desc(rhsShape, dataType, rhsStride);
+    auto dst_md = memory::desc(resShape, dataType, resStride);
+    auto matmul_d = matmul::desc(src_md, weights_md, dst_md);
+    primitive_attr matmul_attr;
+    auto matmul_pd = matmul::primitive_desc(matmul_d, matmul_attr, getEngine());
+    prim.reset(new matmul(matmul_pd));
+    featureSizes.assign(inputSizes, featureSize);
+    std::vector<InferenceEngine::TensorDesc> internalMemDesc = {
```

Comment:
Usage of IE:TensorDesc and IE::Blob is not allowed inside CPU nodes. You should use intel_cpu::Memory and intel_cpu::MemoryDesc instead.